src/mmw/js/src/modeling/views.js

```diff
@@ -142,6 +247,20 @@ var ScenarioTabPanelsView = Marionette.CollectionView.extend({
     },
 
     childView: ScenarioTabPanelView,
+    childEvents: {
+        'rename:scenario': function(view, model, newName) {
+            var match = _.find(this.collection.models, function(model) {
```

Comment:
I think all of this logic could be on a `setName` method on the model, in case we need to rename scenarios from other places.  Should return success so the view can handle the necessary UI